lib/parse_date/int_from_string.rb

```diff
@@ -0,0 +1,140 @@
+    # @return [Integer, nil] Integer year if we could parse one, nil otherwise
+    def self.year_int_from_date_str(orig_date_str)
+      return if orig_date_str == '0000-00-00' # shpc collection has these useless dates
+      # B.C. first in case there are 4 digits, e.g. 1600 B.C.
+      return ParseDate.send(:sortable_year_int_for_bc, orig_date_str) if orig_date_str.match(BC_REGEX)
+
+      result = ParseDate.send(:sortable_year_for_yyyy_or_yy, orig_date_str)
+      result ||= ParseDate.send(:sortable_year_for_decade, orig_date_str) # 19xx or 20xx
+      result ||= ParseDate.send(:sortable_year_for_century, orig_date_str)
+      result ||= ParseDate.send(:sortable_year_int_for_early_numeric, orig_date_str)
+      unless result
+        # try removing brackets between digits in case we have 169[5] or [18]91
+        no_brackets = ParseDate.send(:remove_brackets, orig_date_str)
+        return year_int_from_date_str(no_brackets) if no_brackets
+      end
+      result.to_i if result && year_int_valid?(result.to_i)
+    end
+
+    # true if the year is between -9999 and (current year + 1)
+    # @return [Boolean] true if the year is between -9999 and (current year + 1); false otherwise
+    def self.year_int_valid?(year)
+      return false unless year.is_a? Integer
+
+      (-1000 < year.to_i) && (year < Date.today.year + 2)
+    end
+
+    protected
+
+    # get String sortable value year if we can parse date_str to get a year.
+    # @return [String, nil] String sortable year if we could parse one, nil otherwise
+    #  note that these values must *lexically* sort to create a chronological sort.
+    def sortable_year_for_yyyy_or_yy(orig_date_str)
+      # most date strings have a four digit year
+      result = ParseDate.sortable_year_for_yyyy(orig_date_str)
+      result ||= ParseDate.sortable_year_for_yy(orig_date_str) # 19xx or 20xx
+      result
+    end
+
+    BRACKETS_BETWEEN_DIGITS_REXEXP = Regexp.new('\d[' + Regexp.escape('[]') + ']\d')
+
+    # removes brackets between digits such as 169[5] or [18]91
+    def remove_brackets(orig_date_str)
+      orig_date_str.delete('[]') if orig_date_str.match(BRACKETS_BETWEEN_DIGITS_REXEXP)
+    end
+
+    # looks for 4 consecutive digits in orig_date_str and returns first occurrence if found
+    # @return [String, nil] 4 digit year (e.g. 1865, 0950) if orig_date_str has yyyy, nil otherwise
+    def sortable_year_for_yyyy(orig_date_str)
+      matches = orig_date_str.match(/\d{4}/) if orig_date_str
+      matches&.to_s
+    end
+
+    # returns 4 digit year as String if we have a x/x/yy or x-x-yy pattern
+    #   note that these are the only 2 digit year patterns found in our actual date strings in MODS records
+    #   we use 20 as century digits unless it is greater than current year:
+    #   1/1/15  ->  2015
+    #   1/1/25  ->  1925
+    # @return [String, nil] 4 digit year (e.g. 1865, 0950) if orig_date_str matches pattern, nil otherwise
+    def sortable_year_for_yy(orig_date_str)
```

Comment:
@jcoyne are these the comments you wanted?   I'm happy to have a follow on PR that adds this to the class method and to the README, if that would suffice.